### PR TITLE
Fix stream bracket

### LINKED ISSUE #642: Stream.bracket returns a stream with one element
Following the gitter chat:

```
def readFromFile(filename: String) =
    Stream.bracket(openFile(filename))(closeFile) { in =>
      IO.effect(Option(in.readLine()))
    }
```

returns the Stream with first line of the file, albeit it should repeat until it returns None (in this case, all lines of the file, since null is the end of file). Hence, the following:

```
readFromFile(filename).run(Sink.collect[String].map(_.size))
```

produces 1.

### PR DESCRIPTION
**Pull request by iravid**
Fixes #642.

### CODE REVIEW

File: core/jvm/src/test/scala/scalaz/zio/stream/StreamSpec.scala [hunk 1]
```diff
@@ -374,4 +403,42 @@ class StreamSpec(implicit ee: org.specs2.concurrent.ExecutionEnv)
...
+  private def bracket =
+    unsafeRun(
+      for {
+        done <- Ref.make(false)
+        iteratorStream = Stream.bracket(UIO(Iterator.range(0, 3)))(_ => done.set(true)) { it =>
+          if (it.hasNext) UIO(Some(it.next))
+          else UIO(None)
+        }
+        result <- iteratorStream.run(Sink.collect[Int])
+      } yield result must_=== List(0, 1, 2)
```
Comment (LGLO) [inline core/jvm/src/test/scala/scalaz/zio/stream/StreamSpec.scala]: There is no assertion on value of `done` after stream is run.

File: core/jvm/src/test/scala/scalaz/zio/stream/StreamSpec.scala [hunk 2]
```diff
@@ -374,4 +403,42 @@ class StreamSpec(implicit ee: org.specs2.concurrent.ExecutionEnv)
...
+      for {
+        done <- Ref.make(false)
+        iteratorStream = Stream
+          .bracket(UIO(Iterator.range(0, 3)))(_ => done.set(true)) { it =>
+            if (it.hasNext) UIO(Some(it.next))
+            else UIO(None)
+          }
+          .take(2)
+        result <- iteratorStream.run(Sink.collect[Int])
+      } yield result must_=== List(0, 1)
```
Comment (LGLO) [inline core/jvm/src/test/scala/scalaz/zio/stream/StreamSpec.scala]: There is no assertion on value of `done` after stream is run.

File: core/jvm/src/test/scala/scalaz/zio/stream/StreamSpec.scala (hunk 1 above)
Comment (iravid) [reply core/jvm/src/test/scala/scalaz/zio/stream/StreamSpec.scala]: Right! Thanks.

File: core/jvm/src/test/scala/scalaz/zio/stream/StreamSpec.scala (hunk 2 above)
Comment (iravid) [reply core/jvm/src/test/scala/scalaz/zio/stream/StreamSpec.scala]: 👍

**Pull request merged**